kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -1,263 +1,386 @@
+log_output -i "Set Azure stack environment."
+retrycmd_if_failure 5 10 az cloud set -n $ENVIRONMENT_NAME
+log_output -i "Update cloud profile."
+retrycmd_if_failure 5 10 az cloud update --profile $HYBRID_PROFILE
+
+if [ $IDENTITY_SYSTEM == "ADFS" ] ; then
+	log_output -i "Login to ADFS environment using Azure CLI."	
+	retrycmd_if_failure 5 10 az login --service-principal  -u $SPN_CLIENT_ID  -p $CERTIFICATE_LOCATION --tenant $TENANT_ID --debug 
+else
+	log_output -i "Login to AAD environment using Azure CLI."
+	retrycmd_if_failure 5 10 az login --service-principal -u $SPN_CLIENT_ID -p $SPN_CLIENT_SECRET --tenant $TENANT_ID
 fi
 
-echo "Done building the API model based on the stamp information."
-
-MYDIR=$PWD
-echo "Current directory is: $MYDIR"
-
-echo "Deploy the template using the API model in resource group $MASTER_DNS_PREFIX."
+log_output -i "Setting subscription to $TENANT_SUBSCRIPTION_ID"
+retrycmd_if_failure 5 10 az account set --subscription $TENANT_SUBSCRIPTION_ID > /dev/null
 
-#if AzureAD use below
-#sudo ./bin/acs-engine deploy --resource-group $RESOURCE_GROUP_NAME --azure-env $ENVIRONMENT_NAME --location $REGION_NAME --subscription-id $TENANT_SUBSCRIPTION_ID --client-id $SPN_CLIENT_ID --client-secret $SPN_CLIENT_SECRET --auth-method client_secret --api-model $AZURESTACK_CONFIGURATION
-#if adfs use 
-#sudo ./bin/acs-engine deploy --api-model ./1.11.json --location "local"  --subscription-id e67ebc3e-1604-439d-b5d0-e70235c6ec69 --auth-method "client_certificate" --certificate-path "c:\domain.cer" --private-key-path "c:\domain.key" --azure-env "AzurestackCloud" --client-id "23e22dc8-ad94-4ad0-9b09-e329969f5500" 
-
-if [ $IDENTITY_SYSTEM == "ADFS" ]
-then
-	echo "Using ADFS"
-	sudo ./bin/acs-engine deploy --resource-group $RESOURCE_GROUP_NAME --api-model $AZURESTACK_CONFIGURATION --location $REGION_NAME  --subscription-id $TENANT_SUBSCRIPTION_ID --auth-method client_certificate --certificate-path $CERTIFICATE_LOCATION --private-key-path $KEY_LOCATION --azure-env $ENVIRONMENT_NAME --client-id $SPN_CLIENT_ID 
-else
-	echo "Using AAD"
-	sudo ./bin/acs-engine deploy --resource-group $RESOURCE_GROUP_NAME --azure-env $ENVIRONMENT_NAME --location $REGION_NAME --subscription-id $TENANT_SUBSCRIPTION_ID --client-id $SPN_CLIENT_ID --client-secret $SPN_CLIENT_SECRET --auth-method client_secret --api-model $AZURESTACK_CONFIGURATION
-fi
+log_output -i "Generate ARM template using AKS-Engine."
+retrycmd_if_failure 5 10 sudo ./bin/aks-engine generate $AZURESTACK_CONFIGURATION
 
+log_output -i "ARM template generated at $PWD/_output/$MASTER_DNS_PREFIX. Now changing the directory to arm generated dir."
+cd $PWD/_output/$MASTER_DNS_PREFIX 
 
-echo "Templates output directory is $PWD/_output/$MASTER_DNS_PREFIX"
+log_output -i "Deploy the template."
+az group deployment create -g $RESOURCE_GROUP_NAME --template-file azuredeploy.json --parameters azuredeploy.parameters.json > /dev/null
```

Comment:
I could not find the code to set the azure stack environment and login.

Reply:
Its at Line 359 to 375.